### PR TITLE
Feature/10944 ess processing 307(redirect) fssresponse

### PR DESCRIPTION
**Pull request by ganpat-tech**
PR contains coding changes and automated acceptance test for ess processing 307(redirect) fssresponse.

### CODE REVIEW

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/FunctionalTests/ExcangeSetFssRedirectResponse.cs [hunk 1]
```diff
@@ -0,0 +1,60 @@
+﻿using NUnit.Framework;
+using System.Threading.Tasks;
+using UKHO.ExchangeSetService.API.FunctionalTests.Helper;
+using System.Collections.Generic;
+
+namespace UKHO.ExchangeSetService.API.FunctionalTests.FunctionalTests
+{
+    class ExcangeSetFssRedirectResponse
```
Comment (janvi13603) [inline UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/FunctionalTests/ExcangeSetFssRedirectResponse.cs:8]: Kindly correct typo of class name and file ExcangeSetFssRedirectResponse to ExchangeSetFssRedirectResponse.cs. Also remove line breaks at 17,18

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Helpers/FileShareService.cs [hunk 2]
```diff
@@ -355,6 +357,9 @@ private async Task<bool> ProcessBatchFile(IEnumerable<string> uri, string downlo
                 }
                 HttpResponseMessage httpResponse = await fileShareServiceClient.CallFileShareServiceApi(HttpMethod.Get, payloadJson, accessToken, item, CancellationToken.None, queueMessage.CorrelationId);
 
+                var requestUri = new Uri(httpResponse.RequestMessage.RequestUri.ToString()).GetLeftPart(UriPartial.Path);
+                var serverValue = httpResponse.Headers.Server.ToString().Split('/').First();
```
Comment (janvi13603) [inline UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Helpers/FileShareService.cs:361]: Kindly move this line 361 inside if condition like below
if (httpResponse.IsSuccessStatusCode)
            {
                var serverValue = httpResponse.Headers.Server.ToString().Split('/').First();

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Helpers/FileShareService.cs [hunk 3]
```diff
@@ -395,16 +404,24 @@ public async Task<bool> DownloadReadMeFile(string readMeFilePath, string batchId
             string lineToWrite = string.Concat("File date: ", DateTime.UtcNow.ToString("yyyy-MM-dd HH:mm:ssZ", CultureInfo.InvariantCulture));
             HttpResponseMessage httpReadMeFileResponse;
             httpReadMeFileResponse = await fileShareServiceClient.CallFileShareServiceApi(HttpMethod.Get, payloadJson, accessToken, readMeFilePath, CancellationToken.None, correlationId);
+
+            var requestUri = new Uri(httpReadMeFileResponse.RequestMessage.RequestUri.ToString()).GetLeftPart(UriPartial.Path);
+            var serverValue = httpReadMeFileResponse.Headers.Server.ToString().Split('/').First();
```
Comment (janvi13603) [inline UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Helpers/FileShareService.cs:409]: Kindly move this line 409 inside if condition like below
if (httpReadMeFileResponse.IsSuccessStatusCode)
            {
                var serverValue = httpReadMeFileResponse.Headers.Server.ToString().Split('/').First();

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/FunctionalTests/ExcangeSetFssRedirectResponse.cs (hunk 1 above)
Comment (ganpat-tech) [reply UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/FunctionalTests/ExcangeSetFssRedirectResponse.cs]: Thanks for highlighting this, did changes accordingly.

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Helpers/FileShareService.cs (hunk 2 above)
Comment (ganpat-tech) [reply UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Helpers/FileShareService.cs]: I did changes accordingly.

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Helpers/FileShareService.cs (hunk 3 above)
Comment (ganpat-tech) [reply UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Helpers/FileShareService.cs]: I did changes accordingly.

File: UKHO.SalesCatalogueFileShareServicesMock.API/UKHO.SalesCatalogueFileShareServicesMock.API/Controllers/FileShareServiceController.cs [hunk 4]
```diff
@@ -80,14 +84,20 @@ public IActionResult GetBatches([FromQuery] int? limit, [FromQuery] int start =
...
-        public FileResult DownloadFile(string batchId, string fileName)
+        public ActionResult DownloadFile(string batchId, string fileName)
         {
             byte[] bytes = null;
-            if (!string.IsNullOrEmpty(fileName))
+
+            if (fileName == "DE260001.000")
+            {
+                HttpContext.Response.Headers.Add("Location", fileShareServiceConfiguration.Value.DownloadENCFiles307ResponseUri);
+                return StatusCode((int)HttpStatusCode.RedirectKeepVerb);
```
[the remark this answers is not in the document]
Comment (ganpat-tech) [reply UKHO.SalesCatalogueFileShareServicesMock.API/UKHO.SalesCatalogueFileShareServicesMock.API/Controllers/FileShareServiceController.cs]: Thanks for highlighting this, did changes accordingly.

**Review by janvi13603 — APPROVED** [review]
Thanks for fixing review comments

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.FulfilmentService/Filters/AzureDependencyFilterTelemetryProcessor.cs [hunk 5]
```diff
@@ -0,0 +1,30 @@
+﻿using Microsoft.ApplicationInsights.Channel;
+using Microsoft.ApplicationInsights.DataContracts;
+using Microsoft.ApplicationInsights.Extensibility;
+using System;
+using System.Diagnostics.CodeAnalysis;
+
+namespace UKHO.ExchangeSetService.FulfilmentService.Filters
+{
+    [ExcludeFromCodeCoverage]
```
Comment (barrie-cooper) [inline UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.FulfilmentService/Filters/AzureDependencyFilterTelemetryProcessor.cs:9]: This is perfectly testable so no need to exclude from code coverage and please add some quick unit tests.

Comment (ganpat-tech) [reply UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.FulfilmentService/Filters/AzureDependencyFilterTelemetryProcessor.cs]: Thanks for highlighting, added unit tests cases.

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Helpers/FileShareService.cs [hunk 6]
```diff
@@ -29,6 +29,8 @@ public class FileShareService : IFileShareService
         private readonly ILogger<FileShareService> logger;
         private readonly IFileSystemHelper fileSystemHelper;
         private readonly IMonitorHelper monitorHelper;
+        public const string serverHeaderValue = "Windows-Azure-Blob";
```
[the remark this answers is not in the document]
Comment (ganpat-tech) [reply UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Helpers/FileShareService.cs]: Yes. Did changes accordingly.

**Review by barrie-cooper — APPROVED** [review]
Thanks Ganpat, looks good to go.

**Pull request merged**